### PR TITLE
Feat Sync team deletion

### LINKED ISSUE #7939: 🐛 Bug Report: Membership Deletion Queue Holds Account Deletion
### 👟 Reproduction steps

Discord thread [here](https://discord.com/channels/564160730845151244/636852860709240842/1226918113241600082).

---
Deleted all organizations & then proceeded to delete the account on Cloud, however this [check](https://github.com/appwrite/appwrite/blob/61f04d5340f0fd9afd11e09f7f3ef53278e9882a/app/controllers/api/account.php#L360) returned an error saying there was an active membership attached to the account.

### 👍 Expected behavior

Account should be deleted.

### 👎 Actual Behavior

Account wasn't deleted with the above mentioned error.

### 🎲 Appwrite version

Appwrite Cloud

### 💻 Operating system

Linux

### 🧱 Your Environment

_No response_

### 👀 Have you spent some time to check if this issue has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/appwrite/.github/blob/main/CODE_OF_CONDUCT.md)

### PR DESCRIPTION
**Pull request by byawitz**
## What does this PR do?

Changed team and organization deleting logic to the API instead of queuing it to the deleting worker

## Test Plan

Tested against local `1.5.5` instance, 
1. All of the projects were pulled using the CLI and pushed into a new Organization with different project IDs.
2. A few resources and files were added manually 
3. Delete organization, delete everything.

## Related PRs and Issues

Close #7939 

## Checklist

- [x] Have you read the [Contributing Guidelines on issues](https://github.com/appwrite/appwrite/blob/master/CONTRIBUTING.md)?
- [x] If the PR includes a change to an API's metadata (desc, label, params, etc.), does it also include updated API specs and example docs?

### CODE REVIEW

**Review by christyjacob4 — APPROVED** [review]

**Review by Meldiron — COMMENTED** [review]
Looks good. Now that it's sync, let's update tests for sync behaviour. We either don't have test for this, or it's async (with sleep)

**Comment by byawitz** [comment]
@Meldiron, checked, the test is without `sleep` 

https://github.com/appwrite/appwrite/blob/main/tests/e2e/Services/Teams/TeamsBase.php#L371-L388

File: src/Appwrite/Platform/Workers/Deletes.php [hunk 1]
```diff
@@ -1125,4 +1129,16 @@ function (Document $function) use ($deleteByFunction) {
             );
         }
     }
+
+    /**
+     * @throws Exception
+     * @throws Throwable
+     */
+    public function deleteTeams($dbForConsole, $getProjectDB, $team, $project): void
```
Comment (stnguyen90) [inline src/Appwrite/Platform/Workers/Deletes.php:1137]: Please add typing for these params. Also, this should be singular (`deleteTeam` instead of `deleteTeams`), right?

**Review by stnguyen90 — CHANGES_REQUESTED** [review]

**Comment by stnguyen90** [comment]
> @Meldiron, checked, the test is without `sleep`
> 
> https://github.com/appwrite/appwrite/blob/main/tests/e2e/Services/Teams/TeamsBase.php#L371-L388

@byawitz, but that tests doesn't verify the memberships are deleted synchronously. Maybe you can add a users list memberships? Or at least manually test?

**Comment by byawitz** [comment]
The test returns 404, which means the teams were deleted, and there's no way to delete them in an async way. 

What should the test be different?

**Comment by stnguyen90** [comment]
> The test returns 404, which means the teams were deleted, and there's no way to delete them in an async way.

Even without your change, the current test returns 404 so the current test doesn't actually test anything about memberships? 

> What should the test be different?

A test that verifies the memberships are deleted would be great. I'm not totally sure on how to test, though. Maybe [users.listMemberships()](https://appwrite.io/docs/references/cloud/server-nodejs/users#listMemberships)?

**Comment by byawitz** [comment]
The tests check if the team for a given ID exists. When the endpoint returns `404`, it means the team no longer exists.

Do you mean to add a whole new test for `userMemberships`?

File: src/Appwrite/Platform/Workers/Deletes.php (hunk 1 above)
Comment (byawitz) [reply src/Appwrite/Platform/Workers/Deletes.php:1137]: The method can delete more than one team

**Comment by stnguyen90** [comment]
> Do you mean to add a whole new test for userMemberships

@byawitz, something like that. The goal is to test and make sure the memberships are deleted when the team is deleted.

**Review by stnguyen90 — COMMENTED** [review]
> Do you mean to add a whole new test for userMemberships

@byawitz, something like that. The goal is to test and make sure the memberships are deleted when the team is deleted.

**Comment by byawitz** [comment]
Test added [here](https://github.com/appwrite/appwrite/pull/8217/files#diff-1d6c1e5a3ea8933facab76e4532515a7ff6ed2d9eb896349cbf3e19c4f4cd65a)

File: src/Appwrite/Platform/Workers/Deletes.php [hunk 2]
```diff
@@ -1158,4 +1162,24 @@ function (Document $function) use ($deleteByFunction) {
...
+     * @throws DatabaseException
+     * @throws Exception
+     * @throws Restricted
+     * @throws Structure
+     */
+    public function deleteTeam(Database $dbForConsole, callable $getProjectDB, Document $team, Document $project): void
+    {
+        $this->deleteMemberships($getProjectDB, $team, $project);
+        if ($project->getId() === 'console') {
+            $this->deleteProjectsByTeam($dbForConsole, $getProjectDB, $team);
```
Comment (stnguyen90) [inline src/Appwrite/Platform/Workers/Deletes.php:1182]: Deleting projects synchronously during the API call can take some time, right? Can we avoid that?

File: app/controllers/api/teams.php [hunk 3]
```diff
@@ -338,10 +339,13 @@
     ->label('sdk.response.model', Response::MODEL_NONE)
     ->param('teamId', '', new UID(), 'Team ID.')
     ->inject('response')
+    ->inject('getProjectDB')
     ->inject('dbForProject')
-    ->inject('queueForEvents')
+    ->inject('dbForConsole')
```
Comment (stnguyen90) [inline app/controllers/api/teams.php:344]: `dbForConsole` isn't needed anymore, right?

File: docker-compose.yml [hunk 4]
```diff
@@ -76,6 +76,7 @@ services:
       - appwrite-config:/storage/config:rw
       - appwrite-certificates:/storage/certificates:rw
       - appwrite-functions:/storage/functions:rw
+      - appwrite-builds:/storage/builds:rw
```
Comment (stnguyen90) [inline docker-compose.yml:79]: Is this change needed?

File: app/views/install/compose.phtml [hunk 5]
```diff
@@ -65,6 +65,7 @@ $image = $this->getParam('image', '');
       - appwrite-config:/storage/config:rw
       - appwrite-certificates:/storage/certificates:rw
       - appwrite-functions:/storage/functions:rw
+      - appwrite-builds:/storage/builds:rw
```
Comment (stnguyen90) [inline app/views/install/compose.phtml:68]: Is this change needed?

Comment (byawitz) [reply app/views/install/compose.phtml:68]: Not any more

File: docker-compose.yml (hunk 4 above)
Comment (byawitz) [reply docker-compose.yml:79]: Not any more

**Review by stnguyen90 — APPROVED** [review]

**Review by christyjacob4 — APPROVED** [review]
